### PR TITLE
feat: setup library http

### PR DESCRIPTION
**Pull request by renaldy23**
im adding documentation for using this http library [https://github.com/ibrahimaji/project-novatica/wiki/Dokumentasi-Penggunaan-Library-untuk-Request-API](http)

### CODE REVIEW

File: src/lib/http.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      'Content-Type': 'application/json',
+    },
+  }
+
+  if (Object.keys(body).length > 0) {
+    options.body = body
+  }
+
+  const url = baseurl + endpoint
+  const result = await fetch(url, options)
```
Comment (indrazm) [inline src/lib/http.js:13]: I guess, await fetch should be inside a trycatch block ? Is it ?

File: src/lib/http.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+export const http = async (endpoint, method, body = {}) => {
+  const options = {
+    method: method,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  }
+
+  if (Object.keys(body).length > 0) {
```
Comment (indrazm) [inline src/lib/http.js]: Would do you think if we don't use like this Object.keys()
Instead, we insert into options directly.

```js
const options = {
    method: method,
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body)
  }

```

I guess it's cleaner.

**Review by indrazm — COMMENTED** [review]
It's fine, i only add couple of idea.

File: src/lib/http.js (hunk 2 above)
Comment (renaldy23) [reply src/lib/http.js]: i have some question mas, everytime we send the request it keeps sending body request is it ok ?

**Pull request merged**